opengever/maintenance/scripts/fix_retention_period.py

```diff
@@ -0,0 +1,465 @@
+
+    def apply_recursively(self):
+        if not self.can_apply:
+            return
+
+        if self.apply_to_repo_folder() and self.is_leaf_folder:
+            self.apply_to_dossiers()
+
+        super(RepoFolderDiff, self).apply_recursively()
+
+    def apply_to_dossiers(self):
+        child_dossiers = self.catalog.unrestrictedSearchResults(
+            path='/'.join(self.context.getPhysicalPath()),
+            object_provides=IDossierMarker.__identifier__)
+
+        for brain in child_dossiers:
+            self.apply_to_dossier(brain.getObject())
+
+    def apply_to_repo_folder(self):
+        kind = 'leaf ' if self.is_leaf_folder else ''
+        if self.current_period != DEFAULT_PERIOD:
+            if self.options.verbose:
+                logger.info('skipping {}repo-folder ({}) {}, modified'
+                            .format(kind,
+                                    self.reference_number,
+                                    self.item['_query_path']))
+            return False
+
+        current_title = self.context.Title(prefix_with_reference_number=False)
+        current_title = current_title.decode('utf-8')  # Title returns utf-8
+        xls_title = self.item['effective_title']
+        if current_title != xls_title:
+            if self.options.verbose:
+                logger.info(u'skipping {}repo-folder ({}) {}, title changed '
+                            u'from "{}" to "{}"'
+                            .format(kind,
+                                    self.reference_number,
+                                    self.item['_query_path'],
+                                    xls_title,
+                                    current_title))
+            return False
+
+        self.fixed_folders.add(self.context)
+
+        if self.current_period == self.new_period:
+            if self.options.verbose:
+                logger.info('skipping {}repo-folder ({}) {}, not changed'
+                            .format(kind,
+                                    self.reference_number,
+                                    self.item['_query_path']))
+            return False
+
+        if self.options.verbose:
+            logger.info('fixing {}repo-folder ({}) {}, {}->{}'
+                        .format(kind,
+                                self.reference_number,
+                                self.item['_query_path'],
+                                self.current_period,
+                                self.new_period))
+        self.make_retention_period_backup(self.context, self.current_period)
+        ILifeCycle(self.context).retention_period = self.new_period
+        self.add_journal_entry(self.repo_root, self.context,
+                               self.current_period, self.new_period)
+
+        return True
+
+    def apply_to_dossier(self, dossier):
+        dossier_path = '/'.join(dossier.getPhysicalPath())
+        current_period = ILifeCycle(dossier).retention_period
```

Comment:
`current_period` -> `dossier_period`. The similarity between `current_period` and `self.current_period` is a bit confusing.